api/src/app/modules/attachment/attachment.service.ts

```diff
@@ -92,14 +92,15 @@ export class AttachmentService extends DataService<Attachment, Repository<Attach
   }
 
   async isDeleteAuthorized(entity: Attachment, user?: User):Promise<boolean> {
+    const attachmentTypeCode = entity.attachmentType?.code;
```

Comment:
I don't understand why this change was necessary, maybe it was broken before. AttachmentCreateRequest DTO has attachmentTypeCode, but the AttachmentResponse DTO has attachmentType. The entity has both attachmentTypeCode (the join column) and attachmentType...

Reply:
I suspect if the entity was loaded but not its relations, attachmentType can be null, but attachmentTypeCode should never be null?

Reply:
Actually, while I was testing, the Attachment entity got back has `attachmentType` relation loaded but not the `attachmentTypeCode`. This is strange I found and documented in ticket. So I search the code where has this logic to indirectly use the loaded `attachmentType` to get the `code` so it does not break.

Reply:
The duplicated call on line 167 addCommonRelationsToFindOptions in findFileDatabase is removed but does not fix the loading of `attachmentTypeCode`.
However, during removing, I found the list of "options" actually didn't state "attachmentTypeCode". If I add this `attachmentTypeCode` into the list, it shows up.

But still I chose not to add this `attachmentTypeCode`, it makes sense to me to leave it out since we already specify `attachmentType` relation entity to load so the code is already there as `[ 'id', 'projectId', 'fileName', 'attachmentType']`

Reply:
Replaced old code to use `this.findEntityWithCommonRelations(id)` now.